libfwupdplugin/fu-bluez-device.c

```diff
@@ -293,59 +291,151 @@ fu_bluez_device_get_ble_string_property(const gchar *obj_path,
 	return g_variant_dup_string(val, NULL);
 }
 
+static gchar *
+fu_bluez_device_get_interface_uuid(FuBluezDevice *self,
+				   GDBusObject *obj,
+				   const gchar *obj_path,
+				   const gchar *iface_name)
+{
+	g_autoptr(GDBusInterface) iface = NULL;
+	g_autoptr(GError) error_local = NULL;
+	g_autofree gchar *obj_uuid = NULL;
+
+	iface = g_dbus_object_get_interface(obj, iface_name);
+	if (iface == NULL)
+		return NULL;
+	obj_uuid =
+	    fu_bluez_device_get_ble_string_property(obj_path, iface_name, "UUID", &error_local);
+	if (obj_uuid == NULL)
+		g_debug("failed to get %s property: %s", iface_name, error_local->message);
+
+	return g_steal_pointer(&obj_uuid);
+}
+
+/*
+ * Populates the {uuid_helper : object_path} entry of a device for its
+ * characteristic.
+ */
+static gboolean
+fu_bluez_device_add_characteristic_uuid(FuBluezDevice *self,
+					GDBusObject *obj,
+					const gchar *obj_path,
+					const gchar *iface_name)
+{
+	g_autofree gchar *obj_uuid =
+	    fu_bluez_device_get_interface_uuid(self, obj, obj_path, iface_name);
+
+	if (obj_uuid == NULL)
+		return FALSE;
```

Comment:
if you don't want to set a GError here, maybe return TRUE?

Reply:
In that case the cycle in `fu_bluez_device_ensure_gatt_interfaces()` will not fallback to the next check.

Reply:
Right, I think in that case we should set FWUPD_ERROR_NOT_SUPPORTED and then handle that in the caller with a g_error_matches()